servers/src/main/java/tachyon/master/block/BlockMaster.java

```diff
@@ -647,13 +672,12 @@ public void heartbeat() {
             LOG.error("The worker " + worker + " got timed out!");
             mLostWorkers.add(worker);
             iter.remove();
-          } else if (mLostWorkers.contains(worker)) {
-            LOG.info("The lost worker " + worker + " is found.");
-            mLostWorkers.remove(worker);
```

Comment:
I went again through the codebase and I do believe it does not need this check. More comments at the end of the PR.